src/rest-server/src/utils/frameworkConverter.js

```diff
@@ -0,0 +1,233 @@
+const zlib = require('zlib');
+const axios = require('axios');
+const {Agent} = require('https');
+const _ = require('lodash');
+const yaml = require('js-yaml');
+
+const launcherConfig = require('@pai/config/launcher');
+const {apiserver} = require('@pai/config/kubernetes');
+const k8s = require('@pai/utils/k8sUtils');
+const logger = require('@pai/config/logger');
+
+const decodeName = (name, labels) => {
+  if (labels && labels.jobName) {
+    return labels.jobName;
+  } else {
+    // framework name has not been encoded
+    return name;
+  }
+};
+
+const decompressField = (val) => {
+  if (val == null) {
+    return null;
+  } else {
+    return JSON.parse(zlib.gunzipSync(Buffer.from(val, 'base64')).toString());
+  }
+};
+
+const generateExitDiagnostics = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+
+  let diagnosticsSummary = diag;
+
+  const regex = /matched: (.*)/;
+  const matches = diag.match(regex);
+
+  // No container info here
+  if (matches === null || matches.length < 2) {
+    return diagnosticsSummary;
+  }
+
+  let podCompletionStatus = null;
+  try {
+    podCompletionStatus = JSON.parse(matches[1]);
+  } catch (error) {
+    logger.warn('Get diagnostics info failed', error);
+    return diagnosticsSummary;
+  }
+
+  const summmaryInfo = diag.substring(0, matches.index + 'matched:'.length);
+  diagnosticsSummary = summmaryInfo + '\n' + yaml.safeDump(podCompletionStatus);
+
+  return diagnosticsSummary;
+};
```

Comment:
I remember there are nearly same code at https://github.com/microsoft/pai/blob/41465f33489701db4982b8c8e041383f45ce42db/src/rest-server/src/models/v2/job/k8s.js#L812-L847
Can we reuse this?

Reply:
Agree to reuse code as much as possible. And it also give user same experience on history and current page.

Reply:
There are many detailed changes compared with previous code. Will do code refine and format in another PR, because this might change the behavior of current job API hence postpone release process.